apps/builder/app/dashboard/loading-state/loading.tsx

```diff
@@ -0,0 +1,41 @@
+import { useState } from "react";
+import { Flex, css } from "@webstudio-is/design-system";
+import { useDebounce } from "use-debounce";
+import { SvgLoading } from "~/dashboard/loading-state/svg-loading";
+
+const containerStyle = css({
+  position: "absolute",
+  left: "0",
+  right: "0",
+  top: "0",
+  bottom: "0",
+  background:
+    "radial-gradient(34.37% 50% at 50% 50%, rgba(255, 255, 255, 0.5) 0%, rgba(248, 248, 248, 0.5) 100%);",
+  backdropFilter: "blur(8px)",
+});
+
+export const Loading = ({
+  delay = 0,
+  size,
+}: {
+  delay?: number;
+  size?: number;
+}) => {
+  const [isVisible, setIsVisible] = useState(false);
+  const [visible] = useDebounce(isVisible, delay ?? 0);
+
+  if (!isVisible) {
+    setIsVisible(true);
+  }
```

Comment:
very strange logic, causing a rerender right after the first render just to trigger visible, definitely can be done differently